lib/puppet/functions/hiera_vault.rb

```diff
@@ -100,11 +100,15 @@ def vault_get(key, options, context)
         context.explain { "[hiera-vault] Could not read secret #{path}: #{e.errors.join("\n").rstrip}" }
       end
 
-      next if secret.nil?
+      if secret.nil?
```

Comment:
This is the main new behaviour we care about: when the key is nil, we explain that it's not found, then set the context to .not_found.